packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,494 @@
+        address: "0xUSER",   // The users address
+        keyId: 0,            // OPTIONAL - The Users KeyId they will use
+      },
+      // The Wallets Info
+      provider: {
+        f_type: "ServiceProvider",      // Its a Service Provider
+        f_vsn: "1.0.0",                 // Follows the v1.0.0 spec for service providers
+        address: "0xWallet",            // A flow address owned by the wallet
+        name: "Amazing Wallet",         // OPTIONAL - The name of your wallet. ie: "Dapper Wallet" or "Blocto Wallet"
+        description: "The best wallet", // OPTIONAL - A short description for your wallet
+        icon: "https://___",            // OPTIONAL - Image url for your wallets icon
+        website: "https://___",         // OPTIONAL - Your wallets website
+        supportUrl: "https://___",      // OPTIONAL - An url the user can use to get support from you
+        supportEmail: "help@aw.com",    // OPTIONAL - An email the user can use to get support from you
+      },
+    },
+
+    // Authorization Service
+    {
+      f_type: "Service",
+      f_vsn: "1.0.0",
+      type: "authz",
+      uid: "amazing-wallet#authz",
+      // We will cover this at length in the authorization section of this guide
+    },
+    
+    // User Signature Service
+    {
+      f_type: "Service",
+      f_vsn: "1.0.0",
+      type: "user-signature",
+      uid: "amazing-wallet#user-signature",
+      // We will cover this at length in the user signature section of this guide
+    },
+
+    // OpenID Service
+    {
+      f_type: "Service",
+      f_vsn: "1.0.0",
+      type: "open-id",
+      uid: "amazing-wallet#open-id",
+      method: "DATA",
+      data: { // only include data that was request, ideally only if the user approves the sharing of data, everything is optional
+        f_type: "OpenID",
+        f_vsn: "1.0.0",
+        profile: {
+          name: "bob",
+          family_name: "builder", // icky underscored names because of OpenID Connect spec
+          given_name: "robert",
+          middle_name: "the",
+          nickname: "bob the builder",
+          preferred_username: "bob",
+          profile: "https://www.bobthebuilder.com/",
+          picture: "https://avatars.onflow.org/avatar/bob-the-builder",
+          website: "https://www.bobthebuilder.com",
+          gender: "small cartoonish man",
+          birthday: "1999-01-30", // can use 0000 for year if year is not known
+          zoneinfo: "America/Vancouver", // they are so inconsistent :(
+          locale: "en",
+          updated_at: "1625588304427"
+        },
+        email: {
+          email: "bob@bob.bob",
+          email_verified: false,
+        }
+      },
+    }
+  ]
+}, "*")
+```
+
+### Stoping an Authentication Process.
+
+From any frame, you can send a `FCL:FRAME:CLOSE` post message to FCL, which will halt FCL's current routine and close the frame.
+
+```javascript
+window.parent.postMessage({ type: "FCL:FRAME:CLOSE" }, "*")
```

Comment:
Should these docs include reference to the **wallet-utils** like sendMsgToFcl(), since things like window.parent.postMessage are headed out?